src/python/pants/engine/internals/parser.py

```diff
@@ -122,6 +141,25 @@ def _generate_symbols(
         class Registrar:
             def __init__(self, type_alias: str) -> None:
                 self._type_alias = type_alias
+                for field_type in registered_target_types.aliases_to_types[
+                    type_alias
+                ].class_field_types(union_membership):
+                    registrar_field = RegistrarField(field_type)
+                    setattr(self, field_type.alias, registrar_field)
+                    if field_type.deprecated_alias:
+
+                        def deprecated_field(self):
+                            warn_or_error(
+                                removal_version=field_type.deprecated_alias_removal_version,
+                                entity=f"the field name {field_type.deprecated_alias}",
+                                hint=(
+                                    f"Instead, use `{field_type.alias}`, which behaves the same. Run `{bin_name()} "
+                                    "update-build-files` to automatically fix your BUILD files."
```

Comment:
File an issue to add this behavior to the `build_files` deprecation fixers?

Then yeah drop the auto-fix comment for now.

Reply:
will do

Reply:
> Yeah, I think we can eliminate, since the deprecation warning will be emitted elsewhere anyway?

planning to keep the warn tho, as you may end up with a mix of new vs deprecated field names after applying the build files fixer (until the above mentioned feature has been implemented)